rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -2070,12 +2079,22 @@ public List<JobIdData> reSubmitAll(String sessionId, List<String> jobsId) throws
     }
 
     @Override
+    @RoleWrite
     public boolean pushFile(String sessionId, String spaceName, String filePath, MultipartFormDataInput multipart)
             throws IOException, NotConnectedRestException, PermissionRestException {
         try {
             checkAccess(sessionId, "pushFile");
 
-            Session session = dataspaceRestApi.checkSessionValidity(sessionId);
+            if (Strings.isNullOrEmpty(spaceName)) {
+                throw new IllegalArgumentException("spaceName is empty");
+            }
+
+            Session session = dataspaceRestApi.checkAuthorization(sessionId,
+                                                                  spaceName.toLowerCase().replace("space", ""),
+                                                                  new Object() {
+                                                                  }.getClass().getEnclosingMethod(),
```

Comment:
Is `new Object(){}.getClass().getEnclosingMethod()` meant to get the currently executing method ? (really interesting by the way ^_^) Maybe better to pack it into a variable to be easier to understand?

Reply:
Yes, exactly, that was a trick not easy to find.

I could pack this into a variable all right.